framework/Models/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -80,24 +71,20 @@ def _handleInput(self, paramInput):
       @ In, paramInput, ParameterInput, the already-parsed input.
       @ Out, None
     """
-    PostProcessor._handleInput(self, paramInput)
-    costID  = paramInput.findFirst('costID')
-    self.costID  = costID.value
-    self.invCost = costID.parameterValues['inv']
-
-    valueID = paramInput.findFirst('valueID')
-    self.valueID = valueID.value
-    self.invValue = valueID.parameterValues['inv']
-
-    costLimit  = paramInput.findFirst('costLimit')
-    if costLimit is not None:
-      self.costLimit  = costLimit.value
-      self.costLimitType = costLimit.parameterValues['type']
-
-    valueLimit = paramInput.findFirst('valueLimit')
-    if valueLimit is not None:
-      self.valueLimit = valueLimit.value
-      self.valueLimitType = valueLimit.parameterValues['type']
+    self.objectives = {}
+
+    for child in paramInput.subparts:
+      if child.getName() == 'objective':
+        self.objectives[child.value]={}
+        if child.parameterValues['goal'] in ['min','max']:
+          self.objectives[child.value]['goal'] = child.parameterValues['goal']
+        else:
+          self.raiseAnError(IOError, 'ParetoFrontier postprocessor {}: the objective {} expects a min/max goal, but received {} inputs!".'
+                                  .format(self.name,child.value,child.parameterValues['goal']))
+        if 'upperLimit' in child.parameterValues.keys():
+          self.objectives[child.value]['upperLimit'] = child.parameterValues['upperLimit']
```

Comment:
Maybe you can use child.parameterValues.get('upperLimit')? Default is None if not available

Reply:
edited